### ISSUE TITLE
Pathfinder doesn't work on snow blocks

### ISSUE THREAD

**Issue by MarkuBu** [minetest/minetest#4192]
I noticed that the pathfinder doesn't work if the player stands on a snow block like the one I am pointing on the screenshot.

If I stand on the dirt_with_snow block in front the pathfinder works

That means, just have a little bit snow in your pocket to be safe for mobs

![screenshot_20160606_180024](https://cloud.githubusercontent.com/assets/16163058/15828711/a6512b86-2c10-11e6-9625-d3e2fac2bef2.png)

**Comment by est31** [comment]
Yeah its one of the many problems the pathfinder has. The issue is that if you add in nodeboxes, the pathfinder problem gets even harder.

**Comment by tobyplowy** [comment]
@est31 the Pathfinder really needs to be improved, it has a lot of problems :/

**Comment by est31** [comment]
Doors are the same problem really. You can pass them in some directions but in some you can't. You also can't just special case the opened door because an opened door makes some directions unpassable...

Then there is the issue of slabs on the floor plus slabs on the ceiling... depending on the height of the character this might make it impossible for them to pass... Or not! Or turned slabs right and left that leave a space of a full node inside (but the node boundary is right at the middle of the space, each side contributing a half node of space). You can walk there just fine.

There are solutions for this computational problem: mesh based pathfinders instead of node based ones.

That's why I said that introducing nodeboxes makes the problem so much harder :).

**Comment by nininik0** [comment]
The pathfinder seems to consider any node that is at least has a collision that could be walked on even if it is tiny it is still considered a solid one.
